torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -628,6 +631,71 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
 
         return outputs
 
+    @torch.no_grad()
+    def _write_back_current_shard(self):
+        for p in self.params:
+            if not p._is_sharded:
+                pass
+            chunks = p._full_param_padded.chunk(self.world_size)
```

Comment:
wondering why not calling _get_shard() directly?

Reply:
You're right that no padding will happen if call `self._get_shard(p._full_param_padded)`.

The issue around copying still there. In https://github.com/pytorch/pytorch/blob/master/torch/distributed/_fsdp/fully_sharded_data_parallel.py#L364 you can see `_get_shard` calls `chunk.clone()` unconditionally on the chunk.

Reply:
good point, the change here looks good, thanks!